processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -819,17 +678,21 @@ public Builder setContext(Map<String, Object> context)
       return this;
     }
 
+    public Builder overrideContext(Map<String, Object> contextOverride)
+    {
+      this.context = computeOverriddenContext(context, contextOverride);
+      return this;
+    }
+
     public Builder setHavingSpec(HavingSpec havingSpec)
     {
       this.havingSpec = havingSpec;
-
       return this;
     }
 
     public Builder setLimit(Integer limit)
     {
       this.limit = limit;
```

Comment:
@gianm `setLimit()` was duplicating `limit()`, which had a saner implementation. Removed `setLimit()` and renamed `limit()` to `setLimit()`. Added `postProcessingFn = null` to some methods. Also never skip constructor checks now.

@himanshug if you mean recreate limitSpec every time in Builder.build(), it couldn't be done because limitSpec could be set directly.